File: src/main/java/com/drunkensoftball/api/team/service/TeamService.java
package com.drunkensoftball.api.team.service;

import com.drunkensoftball.api.team.domain.Team;
import org.hibernate.validator.constraints.NotBlank;
import org.springframework.data.domain.Pageable;
import org.springframework.validation.annotation.Validated;

import java.util.List;

@Validated
public interface TeamService {

    Team createTeam(@NotBlank(message = "Authentication token required") String token,
                    @NotBlank(message = "Team name required.") String name);

    List<Team> getTeams(@NotBlank(message = "Authentication token required") String token,
                        Pageable pageable);

    Team getTeam(@NotBlank(message = "Authentication token required") String token,
                 @NotBlank(message = "Team uuid required") String teamUuid);
}

File: src/main/kotlin/com/drunkensoftball/api/team/service/TeamService.kt
package com.drunkensoftball.api.team.service

import com.drunkensoftball.api.team.domain.Team
import com.drunkensoftball.api.user.domain.User
import org.springframework.data.domain.Pageable

interface TeamService {

    fun createTeam(user: User,
                   name: String): Team

    fun getTeams(user: User,
                 pageable: Pageable): List<Team>

    fun getTeam(user: User,
                teamUuid: String): Team

    fun updateTeam(user: User,
                   teamUuid: String,
                   name: String?): Team

    fun deleteTeam(user: User,
                   teamUuid: String)
}
